generator.go

```diff
@@ -87,3 +89,185 @@ func merge(cs ...<-chan error) <-chan error {
 	}()
 	return out
 }
+
+func getRandom(kind string) string {
+	switch kind {
+	case "environment":
+		return environments[rand.Intn(len(environments))]
+	case "organization":
+		return organizations[rand.Intn(len(organizations))]
+	case "role":
+		return roles[rand.Intn(len(roles))]
+	case "platform":
+		return platforms[rand.Intn(len(platforms))]
+	case "tag":
+		return tags[rand.Intn(len(tags))]
+	case "source_fqdn":
+		return sourceFqdns[rand.Intn(len(sourceFqdns))]
+	default:
+		return ""
+	}
+}
+
+func genRandomResourcesTree() []interface{} {
+	resourcesSize := rand.Intn(8)
+	randResources := make([]interface{}, resourcesSize)
+	for i := 0; i < resourcesSize; i++ {
+		randResources[i] = resources[rand.Intn(len(resources))]
+	}
+	return randResources
+}
+
+func genRandomRunList() ([]string, []string) {
+	runListSize := rand.Intn(3)
+	runList := make([]string, runListSize)
+	recipeList := make([]string, runListSize)
+	for i := 0; i < runListSize; i++ {
+		cb := randCookbooks[rand.Intn(len(randCookbooks))]
+		runList[i] = fmt.Sprintf("recipe[%s::default]", cb)
+		recipeList[i] = fmt.Sprintf("%s::default", cb)
+	}
+	return runList, recipeList
+}
+
+func genRandomAttributes() map[string]interface{} {
+	attrSize := rand.Intn(10)
+	randAttributes := make(map[string]interface{}, attrSize)
+	for i := 0; i < attrSize; i++ {
+		k := randAttributeMapKey(attributes)
+		randAttributes[k] = attributes[k]
+	}
+	return randAttributes
+}
+
+func randAttributeMapKey(m map[string]interface{}) string {
+	i := rand.Intn(len(m))
+	for k, _ := range m {
+		if i == 0 {
+			return k
+		}
+		i--
+	}
+	return ""
+}
+
+func randomChefClientRun(chefClient chef.Client, nodeName string) {
+	var (
+		startTime              = time.Now().UTC()
+		endTime                = time.Now().UTC() // TODO: (@afiune) should we mock the time the CCR took to run?
```

Comment:
Can the `endTime` be a few minutes after the `startTime`. The run-history component displays how long the run lasted. The `endTime` could also be a random amount of minutes from 1 to 10 after the `startTime`